core/string/ustring.cpp

```diff
@@ -1157,6 +1157,14 @@ Vector<String> String::split_spaces() const {
 
 Vector<String> String::split(const String &p_splitter, bool p_allow_empty, int p_maxsplit) const {
 	Vector<String> ret;
+
+	if (is_empty()) {
```

Comment:
As written here https://github.com/godotengine/godot/pull/72547#discussion_r1093846393
- An empty p_splitter behaves the same as p_allow_empty set to true.
- So this condition needs to be adjusted: `if (p_allow_empty || p_splitter.is_empty())`

Reply:
I think you're wrong and you're confusing strings with a space in them for empty strings. `allow_empty`. is meant for situations like `"lmoffa".split("f")`. Since there are two f's next to each other, this would normally result in `["lmo", "", "a"]`, but if you turn off `allow_empty`, the empty string won't be there.

It seems like `allow_empty` is only to disqualify empty strings, and this is further evidenced by stuff like `"".split("L")` returning `[""]` if `allow_empty` is true, and `[]` otherwise.

Reply:
Check the logic below: If `p_splitter` is empty, `end` is set to 1. And below it will execute the same code as when `p_allow_empty` is set to `true`.
This is where my thought above comes from.
We may should let other people check this as well, as this is a weird corner case. 😄

Reply:
Here's my reasoning for how it behaves right now (and after my PR):

- The delimiters are the splitting points that divide the string into pieces
- If the string is empty, there are no delimiters in it, so the string remains whole. (return the original empty string)
- If the delimiter is empty, the spaces between characters divide the string into pieces
- If the string is empty, there are not even empty delimiters in it, so the string remains whole. (return the original empty string)
- If the string is not empty, the delimiters are everywhere and don't get removed, unlike other delimiters (since they're empty strings), meaning that indeed, empty strings cannot occur.
- `p_allow_empty` just removes the empty strings if false, nothing else.

Reply:
My thought actually comes from the code:
Let's imagine that there is no infinite loop here, e.g. the exit condition is correct.

**Test 1:**
`p_splitter` = ""
`p_allow_empty` = true

`from` = 0
`end` = 1

`if (p_allow_empty || (end > from)) {...}` -> Both are true.

---

**Test 2:**
`p_splitter` = ""
`p_allow_empty` = false

`from` = 0
`end` = 1

`if (p_allow_empty || (end > from)) {...}` -> `end > from` is true.

---

So in both cases the code would actually execute the code inside the `if (p_allow_empty || (end > from)) {...}`.
Therefore, this solution will not work as if the original code would simply work.

Reply:
But does it produce an unexpected result from a user's perspective?

Reply:
Not sure to be honest. 😄 
All other languages I know do not have something like a `allow_empty` option.
C# has something like this, but it behaves weird.

Well I guess it is fine to just stick with it and see if someone complains.
Since this is a rather weird corner case, it is hard to tell what everyone would expect here. :D